File: SEP3_ChatSystem/Model/Unit/Message/GroupMessage.cs
﻿namespace SEP3_ChatSystem.Model.Unit.Message
{
    public class GroupMessage : Message
    {
        public string GroupId { get; set; }

        public GroupMessage(string senderId, string groupId, string messageInfo) : base(senderId, messageInfo,
            MessageType.GROUP)
        {
            GroupId = groupId;
        }
    }
}

File: SEP3_Database/src/main/java/com/example/SEP3_Database/model/domain/unit/message/GroupMessage.java
package com.example.SEP3_Database.model.domain.unit.message;

import com.example.SEP3_Database.model.domain.unit.time.MyTime;

public class GroupMessage extends Message{
    private String GroupId;

    public GroupMessage(String senderId, String groupId, String messageInfo)
    {
        super(senderId, messageInfo, MessageType.GROUP);
        GroupId = groupId;
    }

    public GroupMessage(String senderId, String groupId, String messageInfo, MyTime time)
    {
        super(senderId, messageInfo, MessageType.PRIVATE,time);
        GroupId = groupId;
    }

    public String getGroupId() {
        return GroupId;
    }

    public void setGroupId(String groupId) {
        GroupId = groupId;
    }

    public GroupMessage copy()
    {
        return new GroupMessage(super.SenderId,GroupId,super.MessageInfo,super.Time);
    }
}
